src/wdqms/wdqms.py

```diff
@@ -487,10 +508,37 @@ def _genqsat(self, df):
             logging.debug(f"Original q background departure: {q_tmp['Obs_Minus_Forecast_adjusted'].values}")
             logging.debug(f"New q background departure: {bg_dep}")
 
-            # Replace the current background departure with the new calculated one
-            df.loc[(df['var_id'] == self.wdqms_type_dict[self.wdqms_type]['variable_ids']['q']) &
-                   (df['Station_ID'] == stn),
-                   'Obs_Minus_Forecast_adjusted'] = bg_dep
+            try:
+                # This method works for nearly all cases.
+                # Find where the current station is in the input dataframe, then where
+                # the var_id == q value, replace the values with the background departure.
+                # Add tmp dataframe to a list to be concatenated into a final dataframe.
+                condition = (df['Station_ID'] == stn)
+
+                tmp = df.loc[condition]
+                tmp.loc[tmp['var_id'] == self.wdqms_type_dict[self.wdqms_type]['variable_ids']['q'], 'Obs_Minus_Forecast_adjusted'] = bg_dep
```

Comment:
There is a warning message at run time:
```
wdqms.py:519 FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas
```
This may cause an error (not warning) in future pandas release.